include/h2o.h

```diff
@@ -1299,6 +1294,11 @@ void h2o_replay_request_deferred(h2o_req_t *req);
  * @param generator the generator
  */
 void h2o_start_response(h2o_req_t *req, h2o_generator_t *generator);
+/**
+ * indicates whether the response is being sent or not
+ * @param req the request
+ */
+static int h2o_is_sending_response(h2o_req_t *req);
```

Comment:
I am going to remove this function, and directly check `req->_generator != NULL` in http1.c (the only code that uses this function).

The name ("is_sending_response") is inaccurate, as generator is assigned before the emission of the response starts and might remain assigned after the generator goes away (IIRC this is the case when there is a buffering ostream filter).

If a generator has been assigned is something that each protocol implementation (e.g., http1.c) can know about, and this information is only necessary to http1.c, due to HTTP/1 (ab)using an HTTP response to indicate the framing error (HTTP/2, /3 reports framing issues using frame-level errors).

Reply:
Done in 829cb56.

Reply:
Actually mruby.c also touches `req->_generator` in several points to know whether it can send the response header or not, and I intended to use this function in there too after this PR gets merged. Touching `req->_generator` makes me a bit anxious, so we should come up with another nice name, or at least removing `_` prefix from `_generator` if we keep touching that member directly, WDYT?